src/app/commands/cmd_sprite_size.cpp

```diff
@@ -319,15 +328,77 @@ bool SpriteSizeCommand::onEnabled(Context* context)
 
 void SpriteSizeCommand::onExecute(Context* context)
 {
+#ifdef ENABLE_UI
+  const bool ui = (params().ui() && context->isUIAvailable());
+#endif
   const ContextReader reader(context);
   const Sprite* sprite(reader.sprite());
-  int new_width = (m_width ? m_width: int(sprite->width()*m_scaleX));
-  int new_height = (m_height ? m_height: int(sprite->height()*m_scaleY));
-  ResizeMethod resize_method = m_resizeMethod;
+  auto& params = this->params();
+
+  double ratio = sprite->width() / double(sprite->height());
+  if (params.scale.isSet()) {
+    params.setWidth(int(sprite->width() * params.scale()));
```

Comment:
```suggestion
    params.width(int(sprite->width() * params.scale()));
```
As we saw above ☝️ we can use the `Param<int>::operator()(const int&)` to change the value of a param (and we can remove the `setWidth()`, etc.)

Reply:
Got it!